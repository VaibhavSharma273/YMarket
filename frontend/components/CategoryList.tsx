import React from 'react';
import { StyleSheet, SectionList, View, Text, FlatList } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { Button } from 'react-native-paper';
import { SafeAreaView } from 'react-native-safe-area-context';
import { RootTabScreenProps } from '../types';

import Category from './Category'

const CATEGORIES = [
    {
        id: '1',
        title: 'general'
    },
    {
        id: '2',
        title: 'clothing'
    },
    {
        id: '3',
        title: 'furniture'
    },
    {
        id: '4',
        title: 'books/textbooks'
    },
    {
        id: '5',
        title: 'electronics'
    },
    {
        id: '6',
        title: 'other'
    },
]

const Item = ({ title }: any) => (
    <View style={styles.item}>
      <Text style={styles.title}>{title}</Text>
    </View>
  );

const CategoryList = ({ navigation }: any) => {
    const renderItems = ({ item }: any) => {
        return <Category category={item} navigation={navigation}/>;
    }

    return (
        <SafeAreaView style={styles.container}>
            <FlatList
                data={CATEGORIES}
                scrollEnabled={false}
                renderItem={renderItems}
                keyExtractor={item => item.id}
            />
        </SafeAreaView>
    )
}

const styles = StyleSheet.create({
    container: {
        justifyContent: "center",
        alignItems: "center",
    },
    list: {
        margin: 10,
        height: "90%",
        width: "100%",
    },
    item: {
        borderRadius: 16,
        backgroundColor: "#fff",
        marginTop: 10,
        marginBottom: 10,
        marginLeft: 20,
        marginRight: 20,
    },
    title: {
      width: "100%",
      marginTop: 20,
      fontSize: 25,
      fontWeight: "bold",
      marginLeft: "10%",
    },
  });  

export default CategoryList;