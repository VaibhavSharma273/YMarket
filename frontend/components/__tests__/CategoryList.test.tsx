import * as React from 'react';
import renderer from 'react-test-renderer';

import CategoryList from '../CategoryList';

it(`renders correctly`, () => {
    const props = {
        navigation: jest.fn()
    };
  const tree = renderer.create(<CategoryList {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
