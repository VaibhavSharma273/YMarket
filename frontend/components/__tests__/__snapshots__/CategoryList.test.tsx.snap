// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "1",
          "title": "general",
        },
        Object {
          "id": "2",
          "title": "clothing",
        },
        Object {
          "id": "3",
          "title": "furniture",
        },
        Object {
          "id": "4",
          "title": "books/textbooks",
        },
        Object {
          "id": "5",
          "title": "electronics",
        },
        Object {
          "id": "6",
          "title": "other",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 16,
              "elevation": 3,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 16,
                "marginBottom": 10,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 10,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "padding": 8,
                  "paddingLeft": 16,
                }
              }
            >
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontSize": 43,
                    },
                    Object {
                      "fontWeight": "600",
                    },
                  ]
                }
              >
                general
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 16,
              "elevation": 3,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 16,
                "marginBottom": 10,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 10,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "padding": 8,
                  "paddingLeft": 16,
                }
              }
            >
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontSize": 43,
                    },
                    Object {
                      "fontWeight": "600",
                    },
                  ]
                }
              >
                clothing
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 16,
              "elevation": 3,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 16,
                "marginBottom": 10,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 10,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "padding": 8,
                  "paddingLeft": 16,
                }
              }
            >
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontSize": 43,
                    },
                    Object {
                      "fontWeight": "600",
                    },
                  ]
                }
              >
                furniture
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 16,
              "elevation": 3,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 16,
                "marginBottom": 10,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 10,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "padding": 8,
                  "paddingLeft": 16,
                }
              }
            >
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontSize": 43,
                    },
                    Object {
                      "fontWeight": "600",
                    },
                  ]
                }
              >
                books/textbooks
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 16,
              "elevation": 3,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 16,
                "marginBottom": 10,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 10,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "padding": 8,
                  "paddingLeft": 16,
                }
              }
            >
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontSize": 43,
                    },
                    Object {
                      "fontWeight": "600",
                    },
                  ]
                }
              >
                electronics
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 16,
              "elevation": 3,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 16,
                "marginBottom": 10,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 10,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "padding": 8,
                  "paddingLeft": 16,
                }
              }
            >
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontSize": 43,
                    },
                    Object {
                      "fontWeight": "600",
                    },
                  ]
                }
              >
                other
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
